src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -146,6 +146,16 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
 
         public override ObjectData GetData(NodeFactory factory, bool relocsOnly)
         {
+            // If this is an instantiated non-canonical generic type, add it to the generic instantiations
+            // hashtable. This has to be done during the relocsOnly phase to ensure that by the time the hashtable
+            // node gets emitted (relocsOnly is false), it contains all entries, since we can't add entries to 
+            // it after it has been saved.
```

Comment:
MetadataManager already gets a callback when an EEType node gets marked for this sort of bookkeeping, so no - this should go there.